### PR TITLE
Feature/validations sign up

### PR DESCRIPTION
**Pull request by Arochawms10**

### CODE REVIEW

File: src/app/screens/AddressInformation/index.js [hunk 1]
```diff
@@ -17,7 +17,12 @@ type Props = {
   navigation: any,
 };
 
-const AddressInformationScreen = () => (
+const validate = (address) => {
+  return address.street && address.state &&
```
Comment (usaqui-albert) [inline src/app/screens/AddressInformation/index.js:21]: You have `address.street` duplicated

File: src/app/screens/AddressInformation/index.js [hunk 2]
```diff
@@ -26,31 +31,60 @@ const AddressInformationScreen = () => (
           <TextSmall weight="bold">Residence Information</TextSmall>
         </ContainerImage>
         <ContainerInput>
-          <TextInput placeholder="Street address" />
+          <TextInput
+            placeholder="Street address"
+            value={addresInfoValue.street}
+            onChange={(e) => handleUpdateAddress({street: e.nativeEvent.text})}
+            />
```
Comment (usaqui-albert) [inline src/app/screens/AddressInformation/index.js:38]: Indentation

File: src/app/screens/AddressInformation/index.js [hunk 3]
```diff
@@ -26,31 +31,60 @@ const AddressInformationScreen = () => (
...
         </ContainerInput>
         <ContainerInput>
-          <TextInput placeholder="Suite or apt." />
+          <TextInput
+          placeholder="Suite or apt."
```
Comment (usaqui-albert) [inline src/app/screens/AddressInformation/index.js:42]: Indentation

File: src/app/screens/AddressInformation/index.js [hunk 4]
```diff
@@ -26,31 +31,60 @@ const AddressInformationScreen = () => (
...
+          value={addresInfoValue.apartment}
+          onChange={
+            (e) => handleUpdateAddress({apartment: e.nativeEvent.text})}
+          />
         </ContainerInput>
         <ContainerInput>
-          <TextInput placeholder="State" />
+          <TextInput
+          placeholder="State"
```
Comment (usaqui-albert) [inline src/app/screens/AddressInformation/index.js:50]: Indentation

File: src/app/screens/AddressInformation/index.js [hunk 5]
```diff
@@ -26,31 +31,60 @@ const AddressInformationScreen = () => (
...
+          value={addresInfoValue.state}
+          onChange={(e) => handleUpdateAddress({state: e.nativeEvent.text})}
+          />
         </ContainerInput>
         <ContainerInput>
-          <TextInput placeholder="City" />
+          <TextInput
+          placeholder="City"
```
Comment (usaqui-albert) [inline src/app/screens/AddressInformation/index.js:57]: Indentation

File: src/app/screens/AddressInformation/index.js [hunk 6]
```diff
@@ -26,31 +31,60 @@ const AddressInformationScreen = () => (
...
+          value={addresInfoValue.city}
+          onChange={(e) => handleUpdateAddress({city: e.nativeEvent.text})}
+          />
         </ContainerInput>
         <ContainerInput>
-          <TextInput placeholder="ZIP Code" />
+          <TextInput
+          keyboardType={'numeric'}
```
Comment (usaqui-albert) [inline src/app/screens/AddressInformation/index.js:64]: Indentation

File: src/app/screens/AddressInformation/index.js [hunk 7]
```diff
@@ -26,31 +31,60 @@ const AddressInformationScreen = () => (
...
   </LinearGradientContainer>
 );
 
 const AddressInformation = ({ navigation }: Props) => {
+  const [addressInfo, changeAddressInfo] = useState({});
+
   const callback = () => navigation.navigate('Annual Income');
   return (
     <ButtomButtomFixed
+      disabled={!(validate(addressInfo)) }
```
Comment (usaqui-albert) [inline src/app/screens/AddressInformation/index.js:81]: Extra space at the end of line 81

File: src/app/screens/AddressInformation/index.js [hunk 8]
```diff
@@ -26,31 +31,60 @@ const AddressInformationScreen = () => (
...
       callback={callback}
       label="Continue"
+      handleUpdateAddress={
+        (item) => changeAddressInfo({ ...addressInfo,...item})
```
Comment (usaqui-albert) [inline src/app/screens/AddressInformation/index.js:85]: Please use the same indentation you use in `AnnualIncome/index.js` line 56

File: src/app/screens/AnnualIncome/index.js [hunk 9]
```diff
@@ -52,20 +53,36 @@ const AnnualIncomeScreen = () => (
...
+            'Please select a status',
+            [{ text: 'OK', onPress: () => {} }],
+            { cancelable: false },
+          );
+        } else {
+          navigation.navigate('EmploymentStatus');
+        }
+      }}
       label="Continue"
+      handleNavigation={(item) => changeNavigation(true)}
```
Comment (usaqui-albert) [inline src/app/screens/AnnualIncome/index.js:85]: This line shoud raise an eslint error, `item` variable is not being used.

File: src/app/screens/Login/index.js [hunk 10]
```diff
@@ -46,7 +46,11 @@ const LogInScreen = ({ navigation }: Props) => {
             <TextInput placeholder="Email" />
           </ContainerInput>
           <ContainerInput>
-            <TextInput secureTextEntry onChange={(e) => changePassword(`${e.nativeEvent.text}`)} placeholder="Password" />
+            <TextInput
+              secureTextEntry
+              onChange={(e) => changePassword(`${e.nativeEvent.text}`)}
+              placeholder="Password"
+              />
```
Comment (usaqui-albert) [inline src/app/screens/Login/index.js:53]: Indentation

File: src/app/screens/Login/index.js [hunk 11]
```diff
@@ -56,7 +60,10 @@ const LogInScreen = ({ navigation }: Props) => {
             ) : null}
           <ContainerSwitch>
             <TextSwitchWhite>{'Use Touch Id ? '}</TextSwitchWhite>
-            <Switch onValueChange={() => changeSwitch(!flag)} value={flag} />
+            <Switch
+              onValueChange={() => changeSwitch(!flag)} value={flag}
+              style={{ transform: [{ scaleX: 1.5 }, { scaleY: 1.5 }] }}
+              />
```
Comment (usaqui-albert) [inline src/app/screens/Login/index.js:66]: Indentation

File: src/app/screens/PersonalInformationForeign/index.js [hunk 12]
```diff
@@ -19,8 +19,12 @@ type Props = {
   navigation: any,
 };
 
+const validate = (user) => {
+  return user.firstName && user.lastName && user.date &&
+    user.nationality && user.phoneCode && user.phoneNumber
+}
 
-const PersonalInformationForeignScreen = () => {
+const PersonalInformationForeignScreen = ({userInfoValue, handleUpdateUser}) => {
```
Comment (usaqui-albert) [inline src/app/screens/PersonalInformationForeign/index.js:27]: This line should raise an eslint error, missing spaces.

File: src/app/screens/PersonalInformationForeign/index.js [hunk 13]
```diff
@@ -44,23 +51,54 @@ const PersonalInformationForeignScreen = () => {
             <TextSmall weight="bold">Personal Information</TextSmall>
           </ContainerImage>
           <ContainerInput>
-            <TextInput placeholder="First Name" />
+            <TextInput
+              placeholder="First Name"
+              value={userInfoValue.firstName}
+              onChange={(e) =>
+                handleUpdateUser({firstName: e.nativeEvent.text})}
+              />
```
Comment (usaqui-albert) [inline src/app/screens/PersonalInformationForeign/index.js:59]: Indentation

File: src/app/screens/PersonalInformationForeign/index.js [hunk 14]
```diff
@@ -44,23 +51,54 @@ const PersonalInformationForeignScreen = () => {
...
           </ContainerInput>
           <ContainerInput>
-            <TextInput placeholder="Last Name" />
+            <TextInput
+              placeholder="Last Name"
+              value={userInfoValue.lastName}
+              onChange={(e) =>
+                handleUpdateUser({lastName: e.nativeEvent.text})}
+              />
```
Comment (usaqui-albert) [inline src/app/screens/PersonalInformationForeign/index.js:67]: Indentation

File: src/app/screens/PersonalInformationForeign/index.js [hunk 15]
```diff
@@ -44,23 +51,54 @@ const PersonalInformationForeignScreen = () => {
...
           </ContainerInput>
           <ContainerInput>
-            <TextInput placeholder="Date of birth" value={date} onFocus={datePicker} />
+            <TextInput
+              placeholder="Date of birth"
+              value={date}
+              onFocus={datePicker}
+              />
```
Comment (usaqui-albert) [inline src/app/screens/PersonalInformationForeign/index.js:74]: Indentation

File: src/app/screens/PersonalInformationForeign/index.js [hunk 16]
```diff
@@ -44,23 +51,54 @@ const PersonalInformationForeignScreen = () => {
...
           </ContainerInput>
           <ContainerInput>
-            <TextInput placeholder="Nationality" />
+            <TextInput
+              placeholder="Nationality"
+              value={userInfoValue.nationality}
+              onChange={(e) =>
+                handleUpdateUser({nationality: e.nativeEvent.text})}
+              />
```
Comment (usaqui-albert) [inline src/app/screens/PersonalInformationForeign/index.js:82]: Indentation

File: src/app/screens/PersonalInformationForeign/index.js [hunk 17]
```diff
@@ -44,23 +51,54 @@ const PersonalInformationForeignScreen = () => {
...
           <ContainerInput>
             <ContainerCodePhoneInput>
-              <TextInput placeholder="+123" />
+              <TextInput
+                keyboardType={'numeric'}
+                placeholder="+123"
+                value={userInfoValue.phoneCode}
+                onChange={(e) =>
+                  handleUpdateUser({phoneCode: e.nativeEvent.text})}
+                />
```
Comment (usaqui-albert) [inline src/app/screens/PersonalInformationForeign/index.js:92]: Indentation

File: src/app/screens/PersonalInformationForeign/index.js [hunk 18]
```diff
@@ -44,23 +51,54 @@ const PersonalInformationForeignScreen = () => {
...
+              value={userInfoValue.nationality}
+              onChange={(e) =>
+                handleUpdateUser({nationality: e.nativeEvent.text})}
+              />
           </ContainerInput>
           <ContainerInput>
             <ContainerCodePhoneInput>
-              <TextInput placeholder="+123" />
+              <TextInput
+                keyboardType={'numeric'}
```
Comment (usaqui-albert) [inline src/app/screens/PersonalInformationForeign/index.js:87]: No need of brackets, just do `keyboardType="numeric"`

File: src/app/screens/PersonalInformationForeign/index.js [hunk 19]
```diff
@@ -44,23 +51,54 @@ const PersonalInformationForeignScreen = () => {
...
+                placeholder="+123"
+                value={userInfoValue.phoneCode}
+                onChange={(e) =>
+                  handleUpdateUser({phoneCode: e.nativeEvent.text})}
+                />
             </ContainerCodePhoneInput>
             <ContainerPhoneInput>
-              <TextInput placeholder="Phone number" />
+              <TextInput
+                keyboardType={'numeric'}
```
Comment (usaqui-albert) [inline src/app/screens/PersonalInformationForeign/index.js:96]: No need of brackets, just do keyboardType="numeric"

File: src/app/screens/SignUp/index.js [hunk 20]
```diff
@@ -1,6 +1,6 @@
 // @flow
-import React from 'react';
-import { ToastAndroid } from 'react-native';
+import React, {useState} from 'react';
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:2]: This line should raise an eslint error, missing spaces

File: src/app/screens/SignUp/index.js [hunk 21]
```diff
@@ -34,7 +34,18 @@ const showToastWithGravityAndOffset = () => {
   );
 };
 
-const SignUpViewScreen = () => (
+const validate = (text) => {
+  console.log(text);
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:38]: Remove `console.log` usage

File: src/app/screens/SignUp/index.js [hunk 22]
```diff
@@ -34,7 +34,18 @@ const showToastWithGravityAndOffset = () => {
...
+  let reg = /^\w+([\.-]?\w+)*@\w+([\.-]?\w+)*(\.\w{2,3})+$/;
+  if (reg.test(text) === false) {
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:40]: Did you try just doing `return reg.test(text);`?

File: src/app/screens/SignUp/index.js [hunk 23]
```diff
@@ -34,7 +34,18 @@ const showToastWithGravityAndOffset = () => {
...
+    return false;
+  }
+  else {
+    return true;
+  }
+}
+
+const SignUpViewScreen = ({passwordValue, passwordConfirmValue, emailValue, handleEmail,handleConfirmPassword,  handlePassword }) => (
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:48]: This line should raise an eslint error, too long line.

File: src/app/screens/SignUp/index.js [hunk 24]
```diff
@@ -44,26 +55,60 @@ const SignUpViewScreen = () => (
         <TextWhite>Create your profile</TextWhite>
       </ContainerTextError>
       <ContainerInput>
-        <TextInput placeholder="Email" />
+        <TextInput
+          value={emailValue}
+          placeholder="Email"
+          onChange={(e) => handleEmail(`${e.nativeEvent.text}`)}
+          />
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:62]: Indentation

File: src/app/screens/SignUp/index.js [hunk 25]
```diff
@@ -44,26 +55,60 @@ const SignUpViewScreen = () => (
...
       </ContainerInput>
       <ContainerInput>
-        <TextInput placeholder="Password" />
+        <TextInput
+          secureTextEntry
+          value={passwordValue}
+          onChange={(e) => handlePassword(`${e.nativeEvent.text}`)}
+          placeholder="Password"
+          />
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:70]: Indentation

File: src/app/screens/SignUp/index.js [hunk 26]
```diff
@@ -44,26 +55,60 @@ const SignUpViewScreen = () => (
...
         </IconContainer>
       </ContainerInput>
       <ContainerInput>
-        <TextInput placeholder="Confirm password" />
+        <TextInput
+          value={passwordConfirmValue}
+          secureTextEntry
+          placeholder="Confirm password"
+          onChange={(e) => handleConfirmPassword(`${e.nativeEvent.text}`)}
+          />
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:81]: Indentation

File: src/app/screens/SignUp/index.js [hunk 27]
```diff
@@ -44,26 +55,60 @@ const SignUpViewScreen = () => (
...
       </ContainerInput>
+
+      {passwordValue !== passwordConfirmValue && passwordValue
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:84]: Use parenthesis to be sure about the order of the logic evaluation.
You can also do,
```
{condition1 && condition2 && (
    ... react node
)}
```

File: src/app/screens/SignUp/index.js [hunk 28]
```diff
@@ -44,26 +55,60 @@ const SignUpViewScreen = () => (
...
-  const callback = () => navigation.navigate('ResidenceStatus');
+  const [passwordConfirm, handleConfirmPassword] = useState('111');
+  const [password, changePassword] = useState('');
+  const [email, changeEmail] = useState('');
+
+
   return (
     <ButtomButtomFixed
-      callback={callback}
+    callback={() => {
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:101]: Indentation

File: src/app/screens/SignUp/index.js [hunk 29]
```diff
@@ -44,26 +55,60 @@ const SignUpViewScreen = () => (
...
+        navigation.navigate('ResidenceStatus');
+    }}
       label="Start"
+      disabled={!(password && email && password === passwordConfirm && validate(email)) }
+      handleConfirmPassword={(item) => handleConfirmPassword(item)}
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:106]: Just do `handleConfirmPassword={handleConfirmPassword}`

File: src/app/screens/SignUp/index.js [hunk 30]
```diff
@@ -44,26 +55,60 @@ const SignUpViewScreen = () => (
...
+      passwordConfirmValue={passwordConfirm}
+      handlePassword={(item) => changePassword(item)}
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:108]: Just do `handlePassword={changePassword}`

File: src/app/screens/SignUp/index.js [hunk 31]
```diff
@@ -44,26 +55,60 @@ const SignUpViewScreen = () => (
...
+      passwordValue={password}
+      handleEmail={(item) => changeEmail(item)}
```
Comment (usaqui-albert) [inline src/app/screens/SignUp/index.js:110]: Just do `handleEmail={changeEmail}`

File: src/app/screens/AddressInformation/index.js (hunk 1 above)
Comment (Arochawms10) [reply src/app/screens/AddressInformation/index.js:21]: I think isn't duplicated in the validation

File: src/app/components/ui/TextInput/styles.js [hunk 32]
```diff
@@ -14,4 +13,8 @@ export const CompleteInput = styled.TextInput.attrs(({ theme, underlineColorAndr
   margin: 0 auto;
   width: 100%;
   flex: 1;
+  borderBottomWidth: 1;
+  borderColor: ${(props) => props.underlineColorAndroid || 'rgba(255,255,255, .21);'};
```
Comment (jhonwms) [inline src/app/components/ui/TextInput/styles.js:17]: i think that we are using hex!

File: src/app/screens/AddressInformation/index.js [hunk 33]
```diff
@@ -26,31 +31,60 @@ const AddressInformationScreen = () => (
...
+            value={addresInfoValue.street}
+            onChange={(e) => handleUpdateAddress({ street: e.nativeEvent.text })}
+          />
         </ContainerInput>
         <ContainerInput>
-          <TextInput placeholder="Suite or apt." />
+          <TextInput
+            placeholder="Suite or apt."
+            value={addresInfoValue.apartment}
+            onChange={
```
Comment (jhonwms) [inline src/app/screens/AddressInformation/index.js:44]: change this method to onChangeText

File: src/app/screens/AddressInformation/index.js [hunk 34]
```diff
@@ -26,31 +31,60 @@ const AddressInformationScreen = () => (
...
+            onChange={(e) => handleUpdateAddress({ zip: e.nativeEvent.text })}
+          />
         </ContainerInput>
       </Container>
     </ScrollView>
   </LinearGradientContainer>
 );
 
 const AddressInformation = ({ navigation }: Props) => {
+  const [addressInfo, changeAddressInfo] = useState({});
```
Comment (jhonwms) [inline src/app/screens/AddressInformation/index.js:76]: please an initial state, if addressInfo is only gonna holds a string please to '' if is gonna hold several elements, for example, {element1, element2}, is better using an initial state with all possible items in this var

File: src/app/screens/AddressInformation/index.js (hunk 1 above)
Comment (usaqui-albert) [reply src/app/screens/AddressInformation/index.js:21]: Don't follow

File: src/app/components/ui/TextInput/styles.js (hunk 32 above)
Comment (Arochawms10) [reply src/app/components/ui/TextInput/styles.js:17]: is for the opacity only in the label, if you have a better way maybe we can check

File: src/app/screens/AddressInformation/index.js (hunk 1 above)
Comment (usaqui-albert) [reply src/app/screens/AddressInformation/index.js:21]: Sorry I just see it :D

**Pull request merged**